CatalogAPI/Controllers/ItemsController.cs

```diff
@@ -0,0 +1,116 @@
+﻿using CatalogAPI.Data;
+using CatalogAPI.Data.Model;
+using CatalogAPI.DTO;
+using CatalogAPI.Extensions;
+using Microsoft.AspNetCore.Mvc;
+namespace CatalogAPI.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ItemsController : ControllerBase
+    {
+        [HttpGet("categoryId/{categoryId}/page/{page}/pageSize/{pageSize}")]
+        public ActionResult GetItems(int categoryId, int page, int pageSize)
+        {
+            var itemDTOs = Context.Repository
+                .Where(x => x.Id == categoryId)
+                .SelectMany(x => x.Items)
+                .Select(x => new ItemDTO()
+                {
+                    Id = x.Id,
+                    Name = x.Name,
+                    Price = x.Price,
+                    CategoryId = x.CategoryId
+                })
+                .ToPagedList(page, pageSize);
+            return Ok(itemDTOs);
+        }
+        [HttpGet("{id}")]
+        public ActionResult GetItem(int id)
+        {
+            var itemDTO = Context.Repository
+                .SelectMany(x => x.Items)
+                .Select(x => new ItemDTO()
+                {
+                    Id = x.Id,
+                    Name = x.Name,
+                    Price = x.Price,
+                    CategoryId = x.CategoryId
+                })
+                .SingleOrDefault(x => x.Id == id);
+            if (itemDTO == null)
+            {
+                return NotFound();
+            }
+            return Ok(itemDTO);
+        }
+        [HttpPost]
+        public ActionResult AddItem(ItemDTO itemDTO)
+        {
+            if (Context.Repository
+                .SelectMany(x => x.Items)
+                .Any(x => x.Id == itemDTO.Id))
+            {
+                return BadRequest();
```

Comment:
Level 2 violation

Reply:
While creating I did not get Id from client and I will create new Guid everytime.